qt-models/filtermodels.cpp

```diff
@@ -39,6 +39,55 @@ void MultiFilterSortModel::setLayout(DiveTripModel::Layout layout)
 
 bool MultiFilterSortModel::showDive(const struct dive *d) const
 {
+	if (! m_filterData.validFilter) {
+		return true;
+	}
+
+	bool shouldShow = true;
+	if (d->visibility < m_filterData.minVisibility || d->visibility > m_filterData.maxVisibility) {
+		shouldShow = false;
+	}
+
+	if (d->rating < m_filterData.minRating || d->rating > m_filterData.maxRating) {
+		shouldShow = false;
+	}
+
+	// TODO: get the preferences for the imperial vs metric data.
+	// ignore the check if it doesn't makes sense.
+	if (d->watertemp.mkelvin < C_to_mkelvin(m_filterData.minWaterTemp) || d->watertemp.mkelvin > C_to_mkelvin((m_filterData.maxWaterTemp))) {
+		shouldShow = false;
+	}
+	if (d->airtemp.mkelvin < C_to_mkelvin(m_filterData.minAirTemp) || d->airtemp.mkelvin > C_to_mkelvin(m_filterData.maxAirTemp)) {
+		shouldShow = false;
+	}
+
+	if (m_filterData.from.isValid() && d->when < m_filterData.from.toTime_t()) {
+		shouldShow = false;
+	}
+
+	if (m_filterData.to.isValid() && d->when > m_filterData.to.toTime_t()) {
+		shouldShow = false;
+	}
+
+	// tags.
+	if (!m_filterData.tags.isEmpty()) {
+		auto dive_tags = get_taglist_string(d->tag_list).split(",");
+		bool found_tag = false;
+		for (const auto& filter_tag : m_filterData.tags) {
+			for (const auto& dive_tag : dive_tags) {
```

Comment:
a) I dislike use of auto here - why not just use the correct type
b) we tie the `&` to the variable, no the type